azure_functions_worker/utils/dependency.py

```diff
@@ -0,0 +1,344 @@
+                is_python_version('3.9') else
+                PYTHON_ISOLATE_WORKER_DEPENDENCIES_DEFAULT
+            )
+        else:
+            use_new = is_true_like(use_new_env)
+
+        if use_new:
+            cls.reload_all_namespaces_from_customer_deps(cx_working_dir)
+        else:
+            cls.reload_azure_google_namespace_from_worker_deps()
+
+    @classmethod
+    def reload_azure_google_namespace_from_worker_deps(cls):
+        """This is the old implementation of reloading azure and google
+        namespace in Python worker directory. It is not actually re-importing
+        the module but only reloads the module scripts from the worker path.
+
+        It is not doing what it is intended, but due to it is already released
+        on Linux Consumption production, we don't want to introduce regression
+        on existing customers.
+
+        Only intended to be used in Linux Consumption scenario.
+        """
+        # Reload package namespaces for customer's libraries
+        packages_to_reload = ['azure', 'google']
+        for p in packages_to_reload:
+            try:
+                logger.info(f'Reloading {p} module')
+                importlib.reload(sys.modules[p])
+            except Exception as ex:
+                logger.info('Unable to reload {}: \n{}'.format(p, ex))
+            logger.info(f'Reloaded {p} module')
+
+        # Reload azure.functions to give user package precedence
+        logger.info('Reloading azure.functions module at %s',
+                    inspect.getfile(sys.modules['azure.functions']))
+        try:
+            importlib.reload(sys.modules['azure.functions'])
+            logger.info('Reloaded azure.functions module now at %s',
+                        inspect.getfile(sys.modules['azure.functions']))
+        except Exception as ex:
+            logger.info('Unable to reload azure.functions. '
+                        'Using default. Exception:\n{}'.format(ex))
+
+    @classmethod
+    def reload_all_namespaces_from_customer_deps(cls, cx_working_dir: str):
+        """This is a new implementation of reloading azure and google
+        namespace from customer's .python_packages folder. Only intended to be
+        used in Linux Consumption scenario.
+
+        Parameters
+        ----------
+        cx_working_dir: str
+            The path which contains customer's project file (e.g. wwwroot).
+        """
+        # Specialized working directory needs to be added
+        working_directory: str = os.path.abspath(cx_working_dir)
+
+        # Switch to customer deps and clear out all module cache in worker deps
+        cls._remove_from_sys_path(cls.worker_deps_path)
+        cls._add_to_sys_path(cls.cx_deps_path, True)
+        cls._add_to_sys_path(working_directory, False)
+        logger.info('Reloaded azure google namespace from customer deps')
+
+    @classmethod
+    def _add_to_sys_path(cls, path: str, add_to_first: bool):
```

Comment:
Similar to the comment made for `_remove_from_sys_path`, a list of tuples would make it a little cleaner to use.